org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/FileEventHandler.java

```diff
@@ -200,6 +300,23 @@ private static boolean isFolderRenameEvent(FileRenameEvent event) {
 		return (oldPath.toFile().isDirectory() || newPath.toFile().isDirectory()) && Objects.equals(oldPath.removeLastSegments(1), newPath.removeLastSegments(1));
```

Comment:
I think we should remove this   Objects.equals(oldPath.removeLastSegments(1), newPath.removeLastSegments(1).
If i want to  rename package  jdt.pkg1 to  com.pkg1,  then   because "Objects.equals(oldPath.removeLastSegments(1), newPath.removeLastSegments(1)" ,   isFolderRenameEvent will return false.

Reply:
When you are renaming package jdt.pkg1 to com.pkg1, it's equivalent to rename package jdt to com from the LSP perspective. The file uri won't contain the sub package in the path when you rename a parent package.

Reply:
Take a look at the use case to rename package from 'jdt.pkg1' to 'com.pkg1', it's a sensible use case. The latest commit has fixed this case.